server/re/src/icatGeneralMS.cpp

```diff
@@ -138,11 +144,14 @@ msiQuota( ruleExecInfo_t *rei ) {
  * \sa none
 **/
 int  msiSetResource( msParam_t* xrescName, ruleExecInfo_t *rei ) {
-    char *rescName;
+    char* rescName{static_cast<char *>(xrescName->inOutStruct)};
 
-    rescName = ( char * ) xrescName->inOutStruct;
+    if (nullptr == rei->doi) {
+        log_msi::error("{}: Cannot set resource, DOI (DataObjInfo) not found.", __func__);
```

Comment:
`"not found."` seems a little misleading.

How do you feel about `"Cannot set resource, no DOI (DataObjInfo) attached to REI."`?

That captures the situation a bit better IMO. Thoughts?

Reply:
Or is that too specific?

Reply:
I'm fine with the more verbose description. Should we 'full name' the REI like we did with DOI?

Reply:
Sure. It wouldn't hurt to put it in parens.

Reply:
Done.